stdlib/bytes.mli

```diff
@@ -173,6 +173,22 @@ val mapi : (int -> char -> char) -> bytes -> bytes
     index (in increasing index order) and stores the resulting bytes
     in a new sequence that is returned as the result. *)
 
+val fold : (char -> 'a -> 'a) -> bytes -> 'a -> 'a
```

Comment:
Sounds good by me. I'm not so happy with array's implementation of the folds though -- using a for loop means a ref and a write barrier. Maybe I'll change array.fold to use a recursive function as well.